File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String str = strs[0];
        if (strs == null || strs.length == 0) return "";
        for(int i = 1;i < strs.length;i++){
            int j = 0;
            while (j < str.length() &&
                   j < strs[i].length() &&
                   str.charAt(j) == strs[i].charAt(j)) {
                j++;
            }

            str = str.substring(0, j);
            if (str.isEmpty()) return "";
            }
        return str;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        prefix = strs[0]
        for j in range(1,len(strs)):
            i = 0
            while i < len(prefix) and i < len(strs[j]) and prefix[i] == strs[j][i]:
                i += 1
            prefix = prefix[:i]
            if prefix == "":
                return ""
        return prefix